src/x86_64/unwind_rule.rs

```diff
@@ -78,61 +63,40 @@ impl UnwindRuleOffsetSpAndPopRegisters {
             }
         }
 
-        if regs.is_empty() && rule.sp_offset_by_8 == 0 {
-            Some(UnwindRuleX86_64::JustReturn)
-        } else if rule.set_registers(&regs) {
-            Some(UnwindRuleX86_64::OffsetSpAndPopRegisters(rule))
+        if regs.is_empty() && sp_offset_by_8 == 0 {
+            Some(Self::JustReturn)
         } else {
-            None
+            let rule = UnwindRuleOffsetSpAndPopRegisters::try_new(sp_offset_by_8, &regs)?;
+            Some(Self::OffsetSpAndPopRegisters(rule))
         }
     }
+}
+
+#[derive(Default, Clone, Copy, Debug, PartialEq, Eq)]
+pub struct UnwindRuleOffsetSpAndPopRegisters {
+    /// The additional stack pointer offset to undo before popping the registers, divided by 8 bytes.
+    pub sp_offset_by_8: u16,
+    /// An encoded ordering of the callee-save registers to pop from the stack, see register_ordering.
+    pub encoded_registers_to_pop: u16,
```

Comment:
I'm not convinced this should be `pub`, but I don't care that much, especially since `register_ordering` is a separate, public thing.